### DISCUSSION TITLE
Maximum amount of armor stands in a block

### DISCUSSION THREAD

**Discussion by hilltty** [PurpurMC/Purpur#440] [__Dismissed__] [answered]
I would like to see the possibility of limiting the maximum number of armor stands in one block, but if there are more of them than the number specified in the configuration, they will drop out as item.
This will fix all possible lag machines associated with armor stands, without cutting too much of its functionality.

**Comment by 12emin34** [comment]
This is more of a plugin thing imo

**Comment by LucidAPs** [comment]
entity-per-chunk-save-limit on paper.yml can semi help with possible lag machines by adding armor stands as long as the chunk is unloaded for at least 10s.

**Comment by hilltty** [comment]
The problem is completely solved by disabling the "armor-stands-tick" parameter in paper.yml, but this deprives the server of the vanilla mechanics of movement, my version will not spoil the vanilla mechanics.

**Comment by hilltty** [comment]
> entity-per-chunk-save-limit on paper.yml can semi help with possible lag machines by adding armor stands as long as the chunk is unloaded for at least 10s.

This is already done, but sometimes people hide them somewhere in spawn chunks.

**Comment by LucidAPs** [comment]
> The problem is completely solved by disabling the "armor-stands-tick" parameter in paper.yml, but this deprives the server of the vanilla mechanics of movement, my version will not spoil the vanilla mechanics.

No you can make a lag machine even with that option, but it will only effect the minecraft client. Chunk ban essentially.

**Comment by hilltty** [comment]
> > The problem is completely solved by disabling the "armor-stands-tick" parameter in paper.yml, but this deprives the server of the vanilla mechanics of movement, my version will not spoil the vanilla mechanics.
> 
> No you can make a lag machine even with that option, but it will only effect the minecraft client. Chunk ban essentially.

By the way, yes, so I hope that Purpur will implement my proposal :)

**Comment by BillyGalbreath** [comment, marked as answer]
This is definitely something that belongs as a plugin. I'm pretty sure a lot of plugins already exist for it, too. If we were to add this feature then someone else is going to want a limit for minecarts. then someone else will want a limit for this, then a limit for that. It's not feasible to make such a large system that would meet everyone's needs, otherwise we'd spend all our time on creating and maintaining this one thing. Best to let people that want to dedicate such time and resources to perfecting it in a plugin ;)
